src/txmempool.cpp

```diff
@@ -861,11 +897,21 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
             const auto& fwdPos = mapSidechains.at(fwd.scId).fwdTransfersSet.find(tx.GetHash());
             assert(fwdPos != mapSidechains.at(fwd.scId).fwdTransfersSet.end());
 
-            //there must be no dangling fwds, i.e. sc creation is either in mempool or in blockchain
+            //there must be no dangling fwds, i.e. sc creation is either in mempool or in blockchain (and not ceased)
             if (!mapSidechains.at(fwd.scId).scCreationTxHash.IsNull())
                 assert(mapTx.count(mapSidechains.at(fwd.scId).scCreationTxHash));
             else
-                assert(pcoins->HaveSidechain(fwd.scId));
+                assert(pcoins->GetSidechainState(fwd.scId) == CSidechain::State::ALIVE);
```

Comment:
This constraint in fact was already true (and unchecked), right?

Reply:
yes